consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/statemachine/IbftRound.java

```diff
@@ -118,7 +121,20 @@ public ConsensusRoundIdentifier getRoundIdentifier() {
    * @param headerTimeStampSeconds the header time stamp seconds
    */
   public void createAndSendProposalMessage(final long headerTimeStampSeconds) {
-    final Block block = blockCreator.createBlock(headerTimeStampSeconds).getBlock();
+
+    // Determine if we're at a protocol spec that requires withdrawals (even if empty) to be present
+    ProtocolSpec protocolSpec =
```

Comment:
This is duplicated several times.  It feels like we can do this in BftBlockCreator by overriding createBlock, checking the withdrawal validator, and switching on that.  If the sequence number is truly needed then making a new createBlock method with both args.  May need to also do it for PkuQbftBlockCreator since there is no shared parent classes.

Reply:
Good thought. I've done some refactoring as `BftBlockCreator` already has access to the parent header required to get the next block number. So it was a little easier than I thought. Now `QbftRound` and `IbftRound` can just continue to call `createBlock(...)` as they did before, and `BftBlockCreator.createBlock(...)` can decide which type of block to actually create. There's a little repetition in `PkiQbftBlockCreator` like you say, but it's definitely tidier now.